app/tests/challenges_tests/test_tasks.py

```diff
@@ -137,3 +147,182 @@ def test_challenge_request_budget_calculation(challenge_request):
+                    "April": [],
+                    "May": [],
+                    "June": [],
+                    "July": [],
+                    "August": [],
+                    "September": [],
+                    "October": [],
+                    "November": [
+                        str(s1.algorithm_image.algorithm.pk),
+                        str(s2.algorithm_image.algorithm.pk),
+                    ],
+                },
+            },
+        ),
+        str(phase2.pk): PhaseStatistics(
+            challenge_title=ch2.title,
+            average_algorithm_job_run_time=datetime.timedelta(
+                seconds=4407, microseconds=272750
+            ),
+            accumulated_algorithm_job_run_time=datetime.timedelta(
+                days=2, seconds=21120, microseconds=1006
+            ),
+            average_submission_compute_cost=13.47,
+            total_phase_compute_cost=53.87,
+            archive_item_count=11,
+            monthly_costs={
+                2021: {
+                    "January": 0,
+                    "February": 0,
+                    "March": 0,
+                    "April": 0,
+                    "May": 0,
+                    "June": 0,
+                    "July": 0,
+                    "August": 0,
+                    "September": 0,
+                    "October": 0,
+                    "November": 0,
+                    "December": 0,
+                },
+                2022: {
+                    "January": 0,
+                    "February": 0,
+                    "March": 0,
+                    "April": 0,
+                    "May": 0,
+                    "June": 0,
+                    "July": 0,
+                    "August": 0,
+                    "September": 0,
+                    "October": 53.87,
+                    "November": 0,
+                },
+            },
+            algorithms_submitted_per_month={
+                2021: {
+                    "January": [],
+                    "February": [],
+                    "March": [],
+                    "April": [],
+                    "May": [],
+                    "June": [],
+                    "July": [],
+                    "August": [],
+                    "September": [],
+                    "October": [],
+                    "November": [],
+                    "December": [],
+                },
+                2022: {
+                    "January": [],
+                    "February": [],
+                    "March": [],
+                    "April": [],
+                    "May": [],
+                    "June": [],
+                    "July": [],
+                    "August": [],
+                    "September": [],
+                    "October": [
+                        str(s3.algorithm_image.algorithm.pk),
+                        str(s4.algorithm_image.algorithm.pk),
+                        str(s5.algorithm_image.algorithm.pk),
+                    ],

```

Comment:
Do we need to keep the pks of each algorithm submitted? The object saved in the cache would be huge. Won't a count do instead?